### PR TITLE
docs(PPDSC-2332): search on doc site

### PR DESCRIPTION
**Pull request by Xin00163**
PPDSC-2332
Build the search feature for doc site with an out-of-the-box solution [Docsearch](https://www.algolia.com/developers/code-exchange/frontend-tools/docsearch/?utm_source=google&utm_medium=cpc_nonbrand&utm_campaign=Developer_Google_EMEA_Search_NB_Blog_Dynamic&utm_content=blog&utm_term=&_bt=571020055142&_bk=&_bm=&_bn=g&_bg=131398377117&gclid=Cj0KCQjw8uOWBhDXARIsAOxKJ2FisPLizNWraEIjwVwrQaUscxFsCO-dvoJ-bOU-tgaCzoHJ-bMsVVUaAknfEALw_wcB) with the spec design provided.

**What**

1. Background - why this is needed
2. What did you do
3. What does the reviewers should expect

**I have done:**
- [x] Written unit tests against changes
- [x] Written functional tests against the component and/or NewsKit site
- [ ] Updated relevant documentation

**I have tested manually:**
- [x] The feature's functionality is working as expected on Chrome, Firefox, Safari and Edge
- [ ] The screen reader reads and flows through the elements as expected.
- [x] There are no new errors in the browser console coming from this PR.
- [ ] When visual test is not added, it renders correctly on different browsers and mobile viewports (Safari, Firefox, small mobile viewport, tablet)
- [ ] The Playground feature is working as expected

**Before:**

**After:**

**Who should review this PR:**

**How to test:**

### CODE REVIEW

**Comment by newskit-engineering** [comment]
You can preview these changes on:
  Documentation Site: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2332-algolia-search-on-newskit
  Storybook: http://ncu-newskit-docs.s3-website-eu-west-1.amazonaws.com/ppdsc-2332-algolia-search-on-newskit/storybook

**Review by mutebg — APPROVED** [review]
Looks good and works good!

**Review by Vanals — APPROVED** [review]

File: site/components/search/index.tsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
+import React from 'react';
```
Comment (mstuartf) [inline site/components/search/index.tsx:1]: Is there a convention to have a `search.tsx` file for the component or is that just in `src`?

File: site/components/search/index.tsx [hunk 2]
```diff
@@ -0,0 +1,262 @@
...
+          Search
+          <Mono overrides={{stylePreset: 'flagSolidPrimary'}}>⌘K</Mono>
+        </Button>
+      </Visible>
+
+      {isOpen &&
+        createPortal(
+          <DocSearchModal
+            appId="COXG0SA519"
+            apiKey="61ea0de79ad0ae0839c9ad502c248267"
```
Comment (mstuartf) [inline site/components/search/index.tsx:90]: I don't think this is sensitive but it feels like something that should be loaded from env variables.

File: site/components/search/index.tsx [hunk 3]
```diff
@@ -0,0 +1,262 @@
...
+        >
+          <IconFilledSearch />
+          Search
+          <Mono overrides={{stylePreset: 'flagSolidPrimary'}}>⌘K</Mono>
+        </Button>
+      </Visible>
+
+      {isOpen &&
+        createPortal(
+          <DocSearchModal
```
Comment (mstuartf) [inline site/components/search/index.tsx:88]: Perhaps this isn't important but no matter what env you are on (local, PR) all the results are for the prod newskit site.

**Review by mstuartf — COMMENTED** [review]
Very cool

**Review by Srinivasan-Ramasamy — APPROVED** [review]

File: site/components/search/index.tsx (hunk 1 above)
Comment (Xin00163) [reply site/components/search/index.tsx:1]: Good point. I will change to search.tsx

File: site/components/search/index.tsx (hunk 2 above)
Comment (Xin00163) [reply site/components/search/index.tsx:90]: I think this should be OK. Seeing both mui and base just hard coded there

File: site/components/search/index.tsx (hunk 3 above)
Comment (Xin00163) [reply site/components/search/index.tsx:88]: This is a comment from James as well. In DocSearch2, there is a way to configure different envs but I didn't see it in V3. We have two more tickets related to doc search. Will add this there to investigate.

File: site/components/search/index.tsx (hunk 2 above)
Comment (mstuartf) [reply site/components/search/index.tsx]: ok

File: site/components/search/index.tsx (hunk 3 above)
Comment (mstuartf) [reply site/components/search/index.tsx]: Yeah if it's on the ticket to investigate in future that's fine.

**Review by mstuartf — APPROVED** [review]

**Review by Vanals — APPROVED** [review]

**Pull request merged**